myapp/app/models/task.rb

```diff
@@ -5,21 +5,29 @@ class Task < ApplicationRecord
   validates :name, presence: true, length: { minimum: 2, maximum: 32 }
   enum priority: { Low: 1, Normal: 2, High: 3 }
   enum status: { TODO: 1, IN_PROGRESS: 2, DONE: 3 }
+  has_many :task_labels, dependent: :destroy
+  has_many :labels, through: :task_labels
 
   scope :search_by_name_or_description, -> (keyword) { where('name LIKE ? OR description LIKE ?', "#{keyword}%", "#{keyword}%") if keyword.present? }
   scope :search_by_status, -> (status) { where(status: status) if status.present? }
   scope :search_by_user_id, -> (user_id) { where(user_id: user_id) }
+  scope :search_by_label, -> (label_id) { where(labels: { id: label_id }) if label_id.present? }
   scope :sortby, lambda { |column, direction|
     if column.blank? || direction.blank?
       order(created_at: :desc)
     else
       order("#{column}": direction.to_s)
     end
   }
-  scope :search, lambda { |user_id:, status:, keyword:, sort:, direction:|
+  scope :include_labels, -> { includes(:labels) }
+  scope :include_task_labels, -> { includes(:task_labels) }
+  scope :search, lambda { |user_id:, status:, keyword:, label_id:, sort:, direction:|
     search_by_user_id(user_id)
     .search_by_status(status)
     .search_by_name_or_description(keyword)
+    .search_by_label(label_id)
     .sortby(sort, direction)
+    .include_labels
+    .include_task_labels
```

Comment:
別途コメントしましたが
スコープにする必要ないかなと感じましたがどうでしょう？

```suggestion
    .includes(:labels, :task_labels)
```

Reply:
上と同じくscopeから消しました！
https://github.com/Fablic/training/commit/d23ea39f863735532caa2de2c8ed051d57128734